File: examples/python/core/steps/validate-order.py
from aws_durable_execution_sdk_python import durable_step, StepContext

@durable_step
def validate_order(step_context: StepContext, order_id: str) -> dict:
    """Validate an order."""
    # Your validation logic here
    return {"order_id": order_id, "valid": True}

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    order_id = event["order_id"]
    validation = context.step(validate_order(order_id))
    return validation

File: examples/java/core/steps/validate-order.java
public class ValidateOrderHandler extends DurableHandler<Map<String, String>, Map<String, Object>> {

    @Override
    public Map<String, Object> handleRequest(Map<String, String> event, DurableContext ctx) {
        var orderId = event.get("order_id");

        var validation = ctx.step("validate-order", Map.class, stepCtx -> {
            // Your validation logic here
            return Map.of("order_id", orderId, "valid", true);
        });

        return validation;
    }
}
